x/bank/keeper/genesis.go

```diff
@@ -8,7 +8,7 @@ import (
 )
 
 // InitGenesis initializes the bank module's state from a given genesis state.
-func (k BaseKeeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
+func (k BaseKeeper) InitGenesis(ctx sdk.Context, genState *types.GenesisState) {
```

Comment:
Why does this need a reference?

Reply:
No need to copy the structure. Moreover the `DefaultGenesisState` already returns a pointer.

Reply:
Yeah but it has very little to do with the rest of the changes in this PR, doesn't it? Plus, it does not really need to be a reference, am I wrong?

Reply:
Is this how all modules are now? I took a look and none of them have nil checks.

Reply:
@alexanderbez  - some of them are using pointers (eg slashing, mint, evidence,... ) . Re nil check - we can't return error :( this will panic like in all other InitGenesis (here as well) where we call a function which returns an error.

@alessio - using pointers for structures is usually more natural for me (doesn't need to copy the structure).

Reply:
> using pointers for structures is usually more natural for me

Sure. This change is out of this PR's scope nonetheless. As a rule of thumb, I'd avoid introducing our personal preferences because each and every developer has their own :+1:

Reply:
@alessio  - do you want me to create a new PR for pointer change? My main motivation for pointer was use of `DefaultGenesisState` which already returns a pointer and a general coding practices, not personal.

Reply:
Anyway - I'm find with removing the pointer if you want :)